ledger/ledger-api-test-tool/src/main/scala/com/daml/ledger/api/testtool/tests/TransactionService.scala

```diff
@@ -806,6 +807,523 @@ class TransactionService(session: LedgerSession) extends LedgerTestSuite(session
+        } yield {
+          assertGrpcError(
+            failure,
+            Status.Code.INVALID_ARGUMENT,
+            "Missing field: requesting_parties")
+        }
+    }
+
+  private[this] val flatTransactionByIdSameAsTransactionStream =
+    LedgerTest(
+      "TXFlatTransactionByIdSameAsTransactionStream",
+      "Expose the same events for each transaction as the output of getTransactions") { context =>
+      for {
+        Vector(alpha, beta) <- context.participants(2)
+        submitter <- alpha.allocateParty()
+        listener <- beta.allocateParty()
+        _ <- alpha.create(submitter, AgreementFactory(listener, submitter))
+        _ <- synchronize(alpha, beta)
+        trees <- alpha.flatTransactions(listener, submitter)
+        _ <- eventually {
+          for {
+            byId <- Future.sequence(
+              trees.map(t => beta.flatTransactionById(t.transactionId, listener, submitter)))
+          } yield {
+            assertEquals(
+              "The events fetched by identifier did not match with the ones on the transaction stream",
+              trees,
+              byId)
+          }
+        }
+      } yield {
+        // Checks performed in the `eventually` block
+      }
+    }
+
+  private[this] val transactionTreeByEventId =
+    LedgerTest(
+      "TXTransactionTreeByEventId",
+      "Expose a visible transaction tree by event identifier") { context =>
+      for {
+        ledger <- context.participant()
+        party <- ledger.allocateParty()
+        dummy <- ledger.create(party, Dummy(party))
+        tree <- ledger.exercise(party, dummy.exerciseDummyChoice1)
+        byId <- ledger.transactionTreeByEventId(tree.rootEventIds.head, party)
+      } yield {
+        assertEquals("The transaction fetched by identifier does not match", tree, byId)
+      }
+    }
+
+  private[this] val invisibleTransactionTreeByEventId =
+    LedgerTest(
+      "TXInvisibleTransactionTreeByEventId",
+      "Do not expose an invisible transaction tree by event identifier") { context =>
+      for {
+        Vector(alpha, beta) <- context.participants(2)
+        party <- alpha.allocateParty()
+        intruder <- beta.allocateParty()
+        dummy <- alpha.create(party, Dummy(party))
+        tree <- alpha.exercise(party, dummy.exerciseDummyChoice1)
+        _ <- synchronize(alpha, beta)
+        failure <- beta.transactionTreeByEventId(tree.rootEventIds.head, intruder).failed
+      } yield {
+        assertGrpcError(failure, Status.Code.NOT_FOUND, "Transaction not found, or not visible.")
+      }
+    }
+
+  private[this] val transactionTreeByEventIdInvalid =
+    LedgerTest(
+      "TXTransactionTreeByEventIdInvalid",
+      "Return INVALID when looking up an invalid transaction tree by event identifier") { context =>
+      for {
+        ledger <- context.participant()
+        party <- ledger.allocateParty()
+        failure <- ledger.transactionTreeByEventId("dont' worry, be happy", party).failed
```

Comment:
😄

Reply:
Literally coming from the existing implementation. :wink: